examples/cvae-flax/main.py

```diff
@@ -0,0 +1,94 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+
+from data import load_dataset
+import matplotlib.pyplot as plt
+from models import BaselineNet, Decoder, Encoder, cvae_guide, cvae_model
+from train_baseline import train_baseline
+from train_cvae import train_cvae
+
+from jax import random
+import jax.numpy as jnp
+
+from numpyro.examples.datasets import MNIST
+
+
+def main(args):
+    train_init, train_fetch = load_dataset(
+        MNIST, batch_size=args.batch_size, split="train", seed=args.rng_seed
+    )
+    test_init, test_fetch = load_dataset(
+        MNIST, batch_size=args.batch_size, split="test", seed=args.rng_seed
+    )
+
+    num_train, train_idx = train_init()
+    num_test, test_idx = test_init()
+
+    baseline = BaselineNet()
+    baseline_params = train_baseline(
+        baseline,
+        num_train,
+        train_idx,
+        train_fetch,
+        num_test,
+        test_idx,
+        test_fetch,
+        n_epochs=args.num_epochs,
+    )
+
+    cvae_params = train_cvae(
+        cvae_model,
+        cvae_guide,
+        baseline_params,
+        num_train,
+        train_idx,
+        train_fetch,
+        num_test,
+        test_idx,
+        test_fetch,
+        n_epochs=args.num_epochs,
+    )
+
+    x_test, y_test = test_fetch(0, test_idx)
+
+    baseline = BaselineNet()
+    _ = baseline.init(random.PRNGKey(0), x_test)
+    recognition_net = Encoder()
+    _ = recognition_net.init(random.PRNGKey(0), x_test, y_test)
+    generation_net = Decoder()
+    _ = generation_net.init(random.PRNGKey(0), jnp.ones((1, 256), dtype=jnp.float32))
```

Comment:
Just curious why we need those init calls?

Reply:
Thanks, didn't notice.